lib/ch/query.ex

```diff
@@ -157,8 +157,9 @@ defimpl DBConnection.Query, for: Ch.Query do
     [_status, headers | _data] = responses
 
     num_rows =
-      if summary = get_header(headers, "x-clickhouse-summary") do
-        %{"written_rows" => written_rows} = Jason.decode!(summary)
+      with summary <- get_header(headers, "x-clickhouse-summary"),
+           {:ok, summary} <- Jason.decode(summary),
+           written_rows when not is_nil(written_rows) <- Map.get(summary, "written_rows") do
```

Comment:
`with` falls through with the unmatched value. I think we might need to change the logic a bit to preserve `num_rows :: integer | nil`

Reply:
Great catch. What do you think of:

```elixir
    num_rows =
      with summary <- get_header(headers, "x-clickhouse-summary"),
           summary <- Jason.decode!(summary),
           written_rows when not is_nil(written_rows) <- Map.get(summary, "written_rows") do
        String.to_integer(written_rows)
      end
```

Reply:
I think it would "crash" on missing `x-clickhouse-summary` which might be OK. The initial goal was to allow missing `x-clickhouse-summary` but ensure it's JSON (raise otherwise). Not sure about how important it is now, though.